src/vaski-hop.js

```diff
@@ -1,16 +1,55 @@
 'use strict';
 
-var allPlayerImageSelectors, allComputerImgs, scissors, scissors2, paper, paper2, well, well2, continueButton, resetButton, playerSelection, computerSelection, playerName;
+var allPlayerImageSelectors, allComputerImgs, scissors, scissors2, paper, paper2, well, well2, continueButton, resetButton, playerSelection, computerSelection, playerName, winningFigure, selections, gameOutcome;
 
 var typeScissors = 1;
 var typePaper = 2;
 var typeWell = 3;
 
+
+function compareSelections(playerSelection, computerSelection) {
+
+	selections = [playerSelection, computerSelection];
+
+	switch (true) {
+		case selections.includes(typeScissors) && selections.includes(typePaper):
+			winningFigure = typeScissors;
+			console.log("winningFigure is " + winningFigure)
+			break;
+		case selections.includes(typePaper) && selections.includes(typeWell):
+			winningFigure = typePaper;
+			break;
+		case selections.includes(typeWell) && selections.includes(typeScissors):
+			winningFigure = typeWell;
+			break;
+		case computerSelection === playerSelection:
+			winningFigure = 'nobody';
+			break;
+	};
+};
+
+function checkWinner() {
+	switch (true) {
+		case winningFigure === playerSelection:
+			gameOutcome = 'playerWins';
+			break;
+		case winningFigure === computerSelection:
+			gameOutcome = 'computerWins';
+			break;
+		case winningFigure === 'nobody':
+			gameOutcome = 'draw';
+			break;
+
+	};
+};
+
+
+
 var allSelectionTypes = [typeScissors, typePaper, typeWell];
+var namesForSelections = ['Scissors', 'Paper', 'Well'];
 
 var setComputerSelection = function () {
 	computerSelection = allSelectionTypes[Math.floor(Math.random() * allSelectionTypes.length)];
```

Comment:
šitoje funkcijoje tu padarei sprendimą dėl šių dalykų:
- kaip susimuliuoti atsitiktinį kompiuterio pasirinkimą
- į kurį konkretų globalų kintamąjį išsaugoti kompiuterio pasirinkimą
- būtent tuo metu, kai skaičiuosi kompiuterio pasirinkimą, atlikti tos reikšmės išsaugojimą - t.y pasirenki išsaugojimo momentą.

Kiekvienas padarytas sprendimas tampa apribojimu, pvz kada tu gali iškviesti šitą funkciją, kur tu gali naudoti reišmę ir kada tu tai gali daryti.
Vadinasi jei vienas iš šitų trijų dalykų keisis pvz dėl to, kad tu pertvarkysi **kitą**  kodą, pvz kitą funkciją kas ir kokia tvarka keisis - tu turėsi keisti ir šitą funkciją. Toks kodas trapus, sunkiau keičiamas, lengviau įvelti bugų.

Keisk į funkciją kuri neliečia globalių kintamųjų tiesiog grąžina reikšmę ir tu esi laisvas iškviesti šitą funkciją kiek tau reikia kartų, kur tau reikia ir kada reikia - nes tokia funkcija nesujauks tavo žaidimo būsenos.
Čia labai svarbu. Skaityk kol suprasi arba klausk kas neaišku.